cohttp/src/transfer_io.ml

```diff
@@ -63,12 +52,17 @@ module Make (IO : S.IO) = struct
         read_line ic >>= function
         | None -> return Done
         | Some chunk_size_hex -> (
-            match parse_chunksize chunk_size_hex with
-            | None -> return Done
-            | Some 0L ->
+            match
+              Http.Private.Parser.parse_chunk_length (chunk_size_hex ^ "\r\n")
+            with
+            | Error Partial ->
+                assert false
+                (* this branch will never be reached here since we feed the full line *)
+            | Error (Msg _) -> return Done
```

Comment:
Why we want to ignore any error here? There are things like invalid characters and so on

Reply:
I see that the previous version wasn't failing either, just wondering why errors are ok to be ignored here.

Reply:
> just wondering why errors are ok to be ignored here.

Also curious about this. `Done` here makes sense to me with respect to informing the IO layers that there won't be any new data in the stream, but maybe we should also report a  parsing failure instead of silently suppressing it so we can respond with a bad_request etc? In the current form, it seems like we are banking on the next request cycle to result in a parse error as we attempt to parse a http request, but that seems less than ideal.

Reply:
I'll also add that i'm not proposing making a behavior change in this PR, as keeping that as a separate PR will make the most sense.

Reply:
I don't mind turning this (and other failures) like it into real failures that abort the request. I'm not sure how feasible this is with the current implementation, but feel free to try this.

Reply:
> I'm not sure how feasible this is with the current implementation, but feel free to try this.

this will definitely need more experimentation, so keeping it as a separate PR for this is definitely the way to go.